src/jvm/main/org/jetbrains/kotlinx/lincheck/util/ObjectGraph.kt

```diff
@@ -41,11 +41,11 @@ import java.util.*
  * @param onArrayElement callback for array elements traversal, accepts `(array, index, elementValue)`.
  *   Returns an object to be traversed next.
  */
-internal inline fun traverseObjectGraph(
+internal fun traverseObjectGraph(
     root: Any,
     traverseStaticFields: Boolean = false,
-    onField: (obj: Any, field: Field, value: Any?) -> Any?,
-    onArrayElement: (array: Any, index: Int, element: Any?) -> Any?,
+    onField: ((obj: Any, field: Field, value: Any?) -> Any?)?,
+    onArrayElement: ((array: Any, index: Int, element: Any?) -> Any?)?,
```

Comment:
Why `onField` and `onArrayElement` can be `null` now?

Reply:
Optimization for methods which do not need one or the other callback. So that I do not traverse arrays when the `onArrayElement` call is not needed (specifically, for `FieldSearchHelper` class)

Reply:
Let's then add typealiases: 

```
internal typealias FieldCallback = ((obj: Any, field: Field, value: Any?) -> Any?)
internal typealias ArrayElementCallback = ((obj: Any, field: Field, value: Any?) -> Any?)
```

and then use `FieldCallback?` and `ArrayElementCallback?` as argument types.

Reply:
done